pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -30,128 +29,111 @@ func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerMan
-		return false
-	}
-	accessors, exist := c.configManager.LuaScriptAccessors()[kind]
-	if !exist {
-		return false
-	}
-	switch operationType {
-	case configv1alpha1.InterpreterOperationAggregateStatus:
-		return len(accessors.GetStatusAggregationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretHealth:
-		return len(accessors.GetHealthInterpretationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretDependency:
-		return len(accessors.GetDependencyInterpretationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretReplica:
-		return len(accessors.GetReplicaResourceLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretStatus:
-		return len(accessors.GetStatusReflectionLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationRetain:
-		return len(accessors.GetRetentionLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationReviseReplica:
-		return len(accessors.GetReplicaRevisionLuaScript()) > 0
-	}
-	return false
+	_, exist := c.getInterpreter(kind, operationType)
+	return exist
 }
 
 // GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
-func (c *ConfigurableInterpreter) GetReplicas(object *unstructured.Unstructured) (int32, *workv1alpha2.ReplicaRequirements, error) {
-	klog.Infof("ConfigurableInterpreter Execute ReviseReplica")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetReplicaResourceLuaScript()) == 0 {
-		return 0, nil, fmt.Errorf("customized interpreter operation GetReplicas for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) GetReplicas(object *unstructured.Unstructured) (replicas int32, requires *workv1alpha2.ReplicaRequirements, enabled bool, err error) {
+	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretReplica)
+	if !enabled {
+		return
 	}
-
-	luaScript := customAccessor.GetReplicaResourceLuaScript
-	klog.Infof("lua script %s", luaScript)
-
-	return 0, nil, nil
+	vm := luavm.VM{UseOpenLibs: false}
+	replicas, requires, err = vm.GetReplicas(object, luaScript)
+	return
 }
 
 // ReviseReplica revises the replica of the given object.
-func (c *ConfigurableInterpreter) ReviseReplica(object *unstructured.Unstructured, replica int64) (*unstructured.Unstructured, error) {
-	klog.Infof("ConfigurableInterpreter Execute ReviseReplica")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetReplicaRevisionLuaScript()) == 0 {
-		return nil, fmt.Errorf("customized interpreter operation ReviseReplica for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) ReviseReplica(object *unstructured.Unstructured, replica int64) (revised *unstructured.Unstructured, enabled bool, err error) {
+	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretStatus)
```

Comment:
Why use `InterpreterOperationInterpretStatus` instead of `InterpreterOperationReviseReplica`?

Reply:
Good catch!